src/common/ceph_time.h

```diff
@@ -259,6 +259,14 @@ namespace ceph {
 	return time_point(seconds(ts.tv_sec) + nanoseconds(ts.tv_nsec));
       }
     };
+
+    template<typename Clock, typename Duration>
+    signedspan difference(
+      typename std::chrono::time_point<Clock, Duration> minuend,
+      typename std::chrono::time_point<Clock, Duration> subtrahend) {
+      return signedspan(minuend.time_since_epoch().count() -
```

Comment:
Actually, I think the biggest problem is resolution. I'll throw something in to fix that that should address the width-worry, too.